### PR TITLE
Support Windows

### PR DESCRIPTION
**Pull request by admorgan**
Tried to run on Windows, but found a couple of issues.

### CODE REVIEW

**Comment by jlebar** [comment]
I'm very sorry, somehow my notifications were entirely disabled for this repository, so I didn't realize there had been any activity.

Looking at this now.

File: clang_format.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def main():
     git_cf_path = os.path.join(this_dir, "git-clang-format")
     print('Formatting ' + ' '.join(sys.argv[1:]))
     subprocess.run(
-        [git_cf_path, "-f", "--binary", str(clang_format_path()), "--"] + sys.argv[1:]
+        [python, git_cf_path, "-f", "--binary", str(clang_format_path()), "--"] + sys.argv[1:]
```
Comment (jlebar) [inline clang_format.py:111]: Where does this variable `python` come from?  When I run this, I get

```
Traceback (most recent call last):
  File "/var/folders/y8/8280zwdj7g76b0gtfjmkv0br0000gp/T/tmpaqn6igb4/repoel_6ps6e/clang_format.py", line 116, in <module>
    main()
  File "/var/folders/y8/8280zwdj7g76b0gtfjmkv0br0000gp/T/tmpaqn6igb4/repoel_6ps6e/clang_format.py", line 111, in main
    [python, git_cf_path, "-f", "--binary", str(clang_format_path()), "--"] + sys.argv[1:]
NameError: name 'python' is not defined
```

**Review by jlebar — COMMENTED** [review]
Thanks again for this PR, and I'm very sorry for missing this for so long.

File: clang_format.py (hunk 1 above)
Comment (admorgan) [reply clang_format.py:111]: It isn't a variable, it is a command and apparently when I copied it to here from my working dir I forgot to include the quotes around it. That is what I get for playing with web editors in github while trying to do real work at the same time.

Comment (jlebar) [reply clang_format.py:111]: OK!  I think the right thing is `sys.executable`; I'll merge a PR with that change and with the change for the temp filesystem issue.

Thank you again for the PR, I feel awful that I didn't see this for so long.

**Pull request closed without merging**